PennMobile/New Group/PAC Code/PacCodeViewController.swift

```diff
@@ -0,0 +1,150 @@
+        
+        setupPacCodeHStack()
+        setupPacCodeInfoLabel()
+    }
+    
+    var digitLabel : UILabel {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.heightAnchor.constraint(equalToConstant: 70).isActive = true
+        label.widthAnchor.constraint(equalToConstant: 70).isActive = true
+        
+        label.textAlignment = .center
+        label.font = UIFont.boldSystemFont(ofSize: 30)
+        
+        label.backgroundColor = .grey6
+        
+        return label
+    }
+
+    func setupPacCodeHStack() {
+        pacCodeHStack.axis = .horizontal
+        pacCodeHStack.distribution = .equalCentering
+        pacCodeHStack.spacing = 10.0
+        
+        for digitLabel in quadDigitLabel {
+            pacCodeHStack.addArrangedSubview(digitLabel)
+        }
+        
+        view.addSubview(pacCodeHStack)
+        pacCodeHStack.translatesAutoresizingMaskIntoConstraints = false
+        pacCodeHStack.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor).isActive = true
+        pacCodeHStack.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+    }
+    
+    func setupPacCodeInfoLabel() {
+        let pacCodeInfoLabel = UILabel()
+        pacCodeInfoLabel.textColor = .labelPrimary
+        pacCodeInfoLabel.font = UIFont.systemFont(ofSize: 35)
+        pacCodeInfoLabel.text = "Your PAC Code is"
+        
+        view.addSubview(pacCodeInfoLabel)
+        pacCodeInfoLabel.translatesAutoresizingMaskIntoConstraints = false
+        pacCodeInfoLabel.bottomAnchor.constraint(equalTo: pacCodeHStack.topAnchor, constant: -30).isActive = true
+        pacCodeInfoLabel.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+    }
+    
+    func refreshPacCode() {
+        if let pacCode = pacCode {
+            var counter = 0
+            for pacCodeDigit in pacCode {
+                quadDigitLabel[counter].text = "\(pacCodeDigit)"
+                counter += 1
+            }
+        } else {
+            for box in quadDigitLabel {
+                box.text = ""
+            }
+        }
+    }
+}
+
+// MARK: - Local Authentication
+extension PacCodeViewController : LocalAuthentication {
+        
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        handleAuthentication(cancelText: "Go Back", reasonText: "Authenticate to see your PAC Code")
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        pacCode = nil
+        refreshPacCode()
+    }
+    
+    func handleAuthenticationSuccess() {
+        if let pacCode = getPacCode() {
+            self.pacCode = pacCode
+        } else {
+            self.showAlert(withMsg: "Please login to use this feature", title: "Login Error", completion: { self.navigationController?.popViewController(animated: true)} )
```

Comment:
This alert doesn't make much sense. The user has successfully authenticated with FaceID or passcode but the pac code isn't there. That doesn't mean the user isn't logged in though. It just means we don't have the passcode stored locally on the device. We should be fetching the passcode from CampusExpress here instead.